C++/test/algorithm/string/heaps_algorithm.cpp

```diff
@@ -0,0 +1,41 @@
+#define CATCH_CONFIG_MAIN
+#include "third_party/catch.hpp"
+#include "algorithm/string/heaps_algorithm.hpp"
+
+
+TEST_CASE("Base cases", "[string][heaps_algorithm]") {
+    vector<string> output_1;
+    string s = ""; // empty string
+    vector<string> permutations = {""};
+    REQUIRE(heaps_algorithm(s.length(), s, output_1) == permutations);
+    
+    vector<string> output_2;
+    s = "a";
+    permutations = {"a"};
+    REQUIRE(heaps_algorithm(s.length(), s, output_2) == permutations);
+    
+    vector<string> output_3;
+    s = " ";
+    permutations = {" "};
+    REQUIRE(heaps_algorithm(s.length(), s, output_3) == permutations);
+}
+
+
+TEST_CASE("short cases", "[string][heaps_algorithm]") {
+    vector<string> output_4;
+    string s = "abc";
+    vector<string> permutations = {"abc", "bac", "cab", "acb", "bca", "cba"};
+    REQUIRE(heaps_algorithm(s.length(), s, output_4) == permutations);
+
+    vector<string> output_5;
+    s = "a$1";
+    permutations = {"a$1", "$a1", "1a$", "a1$", "$1a", "1$a"};
+    REQUIRE(heaps_algorithm(s.length(), s, output_5) == permutations);
+}
+
+TEST_CASE("long cases", "[string][heaps_algorithm]") {
```

Comment:
I know that it's pretty onerous to write out these longer cases, but it would be nice to have another one or two.

Reply:
OK

Reply:
"Algos" was a nice touch here. 🙂